utils/split_inouts.py

```diff
@@ -0,0 +1,262 @@
+#!/usr/bin/env python3
+"""
+This script provides a temporary solution for the problem of inout top level
+port representation of the BLIF format.
+
+The script loads the design from a JSON file generated by Yosys. Then it splits
+all inout ports along with their nets and connections to cell ports into two.
+Suffixes are automatically added to distinguish between the input and the
+output part.
+
+For example in the given design (in verilog):
+
+ module top(
+  input  A,
+  output B,
+  inout  C
+ );
+
+  IOBUF iobuf (
+   .I(A),
+   .O(B),
+   .IO_$inp(C),
+   .IO_$out(C)
+  );
+
+ endmodule
+
+the resulting design would be:
+
+ module top(
+  input  A,
+  output B,
+  input  C_$inp,
+  output C_$out
+ );
+
+  IOBUF iobuf (
+   .I(A),
+   .O(B),
+   .IO_$inp(C_$inp),
+   .IO_$out(C_$out)
+  );
+
+ endmodule
+
+"""
+import argparse
+import os
+import simplejson as json
+
+# =============================================================================
+
+
+def find_top_module(design):
+    """
+    Looks for the top-level module in the design. Returns its name.
+    """
+
+    for name, module in design["modules"].items():
+        attrs = module["attributes"]
+        if "top" in attrs and attrs["top"] == 1:
+            return name
+
+    return None
+
+
+def get_nets(bits):
+    """
+    Returns a set of numbers corresponding to net indices.
```

Comment:
This function is very weird.  Can you provide some doctest examples?

Reply:
Added docstring and doctest.